### ISSUE TITLE
Why prefer reference equality for validation?

### ISSUE THREAD

**Issue by mjgpy3** [eulerfx/DDDInventoryItemFSharp#6]
Your example rocks! I'm just curious, why does the validation logic use referential equality and not structural?

Thanks,

**Comment by eulerfx** [comment]
This is only for the null check.

**Comment by mjgpy3** [comment]
Thank you! So that's just guarding against `null` somehow bleeding into the system (via .NET or EventStore)?

**Comment by eulerfx** [comment]
Yes, in this case for strings, which are nullable unfortunately.
